Plugins/Tweaks/UnhardcodeShields.cpp

```diff
@@ -10,25 +9,37 @@
 #include "API/CNWSItem.hpp"
 #include "API/CNWSRules.hpp"
 #include "API/CTwoDimArrays.hpp"
-#include "API/Functions.hpp"
 
 #include <cmath>
 
+
 namespace Tweaks {
 
 using namespace NWNXLib;
 using namespace NWNXLib::API;
 
-static Hooks::Hook s_ComputeArmourClass;
 
-UnhardcodeShields::UnhardcodeShields()
+static Hooks::Hook s_ComputeArmorClassHook;
+static Hooks::Hook s_ComputeArmourClassHook;
+
+static int32_t ComputeArmorClassHook(CNWSItem*);
+static void ComputeArmourClassHook(CNWSCreature*, CNWSItem*, BOOL, BOOL);
+
+void UnhardcodeShields() __attribute__((constructor));
+void UnhardcodeShields()
 {
-    static auto s_ReplacedFunc = Hooks::HookFunction(Functions::_ZN8CNWSItem17ComputeArmorClassEv, (void*)&CNWSItem__ComputeArmorClass, Hooks::Order::Final);
-    s_ComputeArmourClass = Hooks::HookFunction(Functions::_ZN12CNWSCreature18ComputeArmourClassEP8CNWSItemii,
-                                        (void*)&CNWSCreature__ComputeArmourClass, Hooks::Order::Late);
+    if (!Config::Get<bool>("UNHARDCODE_SHIELDS", false))
+        return;
+
+    LOG_INFO("Using baseitems.2da to define shield AC and create shield-like items");
+
+    s_ComputeArmorClassHook = Hooks::HookFunction(Functions::_ZN8CNWSItem17ComputeArmorClassEv,
+                                                  (void*)&ComputeArmorClassHook, Hooks::Order::Final);
+    s_ComputeArmourClassHook = Hooks::HookFunction(Functions::_ZN12CNWSCreature18ComputeArmourClassEP8CNWSItemii,
```

Comment:
Could we use more descriptive names for these two hooks? :D

Reply:
But one has a `u` and one doesn't, isn't that descriptive enough?!